saltproc/input_schema.json

```diff
@@ -85,81 +81,236 @@
+                                                "enum": ["cram16", "cram48"]},
+                                            "n_steps": {
+                                                "description": "Number of stochastic iterations for stochastic integrators",
+                                                "type": "number",
+                                                "minimum": 1}
+                                        },
+                                        "default": {}
+                                    }
+                                }
                             }
                         }
                     }
                 }
-
-            ],
-            "required": ["codename", "exec_path", "template_input_file_path", "npop", "active_cycles","inactive_cycles", "geo_file_paths"]
+            ]
         },
-       "simulation": {
-           "description": "Simulation class input parameters",
-           "type": "object",
-           "properties": {
-               "sim_name": {
-                   "description": "Name of simulation",
-                   "type": "string"},
-               "db_name": {
-                   "description": "Output HDF5 database file name",
-                   "type": "string",
+        "simulation": {
+            "description": "Simulation class input parameters",
+            "type": "object",
+            "properties": {
+                "sim_name": {
+                    "description": "Name of simulation",
+                    "type": "string"},
+                "db_name": {
+                    "description": "Output HDF5 database file name",
+                    "type": "string",
+                    "default": "saltproc_results.h5",
                     "pattern": "^(.*)\\.h5$"},
-               "restart_flag": {
-                   "description": "Restart simulation from the step when it stopped?",
-                   "type": "boolean"},
-               "adjust_geo": {
-                   "description": "switch to another geometry when keff drops below 1?",
-                   "type": "boolean"}
-           },
-           "requires": ["sim_name", "db_name", "restart_flag", "adjust_geo"]
-       },
-       "reactor": {
-           "description": "Reactor class input parameters",
-           "type": "object",
-           "properties": {
-               "volume": {
-                   "description": "reactor core volume [cm^3]",
-                   "type": "number",
+                "restart_flag": {
+                    "description": "Restart simulation from the step when it stopped?",
+                    "type": "boolean",
+                    "default": false},
+                "adjust_geo": {
+                    "description": "switch to another geometry when keff drops below 1?",
+                    "type": "boolean",
+                    "default": false}
+            },
+            "default": {},
+            "required": ["sim_name", "db_name"]
```

Comment:
How come `db_name` is still required? It is given a default value

Reply:
Good point.